src/components/TreebankService/TreebankService.js

```diff
@@ -32,18 +35,31 @@ class TreebankService extends Component {
   }
 
   messageHandler(request, responseFn) {
+    const { arethusa } = this.props;
     const { body } = request;
     const [name] = Object.keys(body);
 
-    switch (name) {
-      case 'gotoSentence':
-        this.setState({ redirectTo: body[name].sentenceId });
-
-        responseFn(ResponseMessage.Success(request, { status: 'success' }));
-        break;
+    try {
+      switch (name) {
+        case 'gotoSentence':
+          this.setState({ redirectTo: body.gotoSentence.sentenceId });
 
-      default:
-        responseFn(error(`Unsupported request: ${name}`));
+          responseFn(ResponseMessage.Success(request, { status: 'success' }));
+          break;
+        case 'getMorph':
+          responseFn(ResponseMessage.Success(
+            request,
+            arethusa.getMorph(body.getMorph.sentenceId, body.getMorph.wordId),
```

Comment:
What would be the best way to test that the messaging works?

Reply:
do you mean in a unit test? 

You would need a document or a mock document to test it, and it should return something like 
```
{"RDF":{"Annotation":{"about":"urn:uuid:15b105c9-79f1-47f7-8906-805b20b5640f","creator":{"Agent":{"about":""}},"created":{"$":""},"rights":{"$":""},"hasTarget":{"Description":{"about":""}},"hasBody":{"resource":"urn:uuid:7926cd34-dd80-4139-a3ac-0888b32ebaf5"},"Body":{"about":"urn:uuid:7926cd34-dd80-4139-a3ac-0888b32ebaf5","type":{"resource":"cnt:ContentAsXML"},"rest":{"entry":{"dict":{"hdwd":{"lang":"grc","$":"περί"}},"infl":{"pofs":{"$":"preposition"},"term":{"form":{"$":"περὶ"}}}}}}}}}
```

Reply:
I wasn't sure how to set up an app that would contain this app in an iFrame with `alpheios-messaging`, like how it will eventually be used in Alpheios. Originally I used the [demo client](https://alpheios-project.github.io/arethusa-demo-client/) from your PR, but I think the version of `alpheios-messaging` has changed since then.

Reply:
I did test it in my little demo client (https://alpheios-project.github.io/arethusa-demo-client/) and it appears to work

Reply:
I just updated the demo client this morning

Reply:
That's great, thank you!